deploy/deploy-gcs.yml

```diff
@@ -199,10 +199,33 @@
           delay: 10
           retries: 50
 
-    - name: GCS | Prometheus Objects | Deploy services, ServiceMonitor and Prometheus Object
-      kube:
-        kubectl: "{{ kubectl }}"
-        file: "{{ manifests_dir }}/gcs-prometheus-bundle.yml"
+    - name: GCS | Prometheus Objects 
+      block:
+        - name: Check if the Custrom Resource Definitions are set
+          command: "{{ kubectl }} get customresourcedefinitions servicemonitors.monitoring.coreos.com"
+          register: result
+          until: result.rc == 0
+          delay: 10
+          retries: 30
+
+        - name: Check if Service Monitor CRD is registered
+          command: "{{ kubectl }} get servicemonitors --all-namespaces"
```

Comment:
why this is checked against `--all-namespaces`?

Reply:
In a multi-namespace environment service monitors can be created by the user in any namespace rather than the monitoring namespace as per the team's wish.  I will update it since only one namespace is needed (i.e ```monitoring```) in the current deployment.